### PR TITLE
[author][new] Add datalist-polyfill w/ npm auto-update via single package.json

### PR DESCRIPTION
**Pull request by mfranzke**
Pull request for issue: #12813 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/mfranzke/datalist-polyfill/
 * NPM package url (optional): https://www.npmjs.com/package/datalist-polyfill
 * License and its reference: MIT
 * GitHub popularity (required):
   - Count of watchers: 8
   - Count of stars: 65
   - Count of forks: 24
 * NPM download stats (optional):
   - weekly Downloads: 1795
   - weekly Downloads at its highest so far: 3618
   - total Downloads: 73k
   - jsDelivr hits/day: 70-80k
 * NPM downloads:
The downloads currently vary a little bit - yarn reported 11k the latest: https://yarnpkg.com/en/package/datalist-polyfill

# Essential checklist
 * [x] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] Doesn't already exist / has duplicate issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
 * [x] The parent of the commit(s) in the PR is not older than 3 days.
 * [x] Pull request is sent from a non-master branch with a meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/datalist-polyfill/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "javascript",
+    "html",
+    "autosuggest",
+    "autosuggester",
+    "suggest",
+    "suggester",
+    "select"
+  ],
+  "npmName": "datalist-polyfill",
+  "npmFileMap": [
```
Comment (mfranzke) [inline ajax/libs/datalist-polyfill/package.json]: I do hope that these are filled out correctly - it's mainly the two JS files that would need to get hosted on a CDN for providing those to the users. And those are stored in the root of the project right away.

Comment (sashberd) [inline ajax/libs/datalist-polyfill/package.json]: Why do you use npm? On Git number of releases more than on npm.

File: ajax/libs/datalist-polyfill/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    {
+      "basePath": "",
+      "files": [
+        "*.js"
```
Comment (sashberd) [inline ajax/libs/datalist-polyfill/package.json]: better to use `datalist-polyfill*`
Also in old versions like in 1.0.0 there are also css files that should be hosted

File: ajax/libs/datalist-polyfill/package.json (hunk 1 above)
Comment (mfranzke) [reply ajax/libs/datalist-polyfill/package.json]: It‘s mainly about two of the earliest releases, that included both a typo and adding the package.json for npm usage - those wouldn‘t be necessary to reference any more, as they are way behind regarding functionality and bugfixes.

And as you seem to favor npm over github I‘ve than chosen that one to inlcude for auto-update.

File: ajax/libs/datalist-polyfill/package.json (hunk 2 above)
Comment (mfranzke) [reply ajax/libs/datalist-polyfill/package.json]: Valid point ! But similar to the above: I‘ve deprecated the CSS dependency in the early days of the project, so this version wouldn‘t be necessary any more for reference. Additionally it shouldn‘t necessarly be referenced at all, but included in their project ecosystem directly, as it‘s content is too small, so it’s „cheaper“ in relation to the savings of an extra HTTP request (even from a CDN).

**Comment by mfranzke** [comment]
@sashberd, thanks a lot for your valuable feedback! I‘ve replied to both of your comments and would like to suggest to leave the file as provided.

**Comment by sashberd** [comment]
@extend1994 @sufuf3 See the authors remarks above my review. At your discretion.

File: ajax/libs/datalist-polyfill/package.json (hunk 2 above)
Comment (mfranzke) [reply ajax/libs/datalist-polyfill/package.json]: Hi @extend1994,
thanks a lot for your feedback - and totally agreed, your mentioned change makes perfectly sense to me. I've updated the PR.

**Pull request merged**